analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UriShouldNotBeHardcoded.CSharp12.cs

```diff
@@ -1,7 +1,10 @@
-﻿class PrimaryConstructor(string ctorParam = "file://blah.txt") // FN
+﻿class PrimaryConstructor(string ctorParamUri = "file://blah.txt", // Noncompliant
+                         string ctorParam = "file://blah.txt") // Compliant
 {
-    void Method(string methodParam = "file://blah.txt") // FN
+    void Method(string methodParamUri = "file://blah.txt", // Noncompliant
+                string methodParam = "file://blah.txt") // Compliant
     {
-        var lambda = (string lambdaParam = "file://blah.txt") => lambdaParam; // FN
+        var lambda = (string lambdaParamUri = "file://blah.txt") => lambdaParamUri; // Noncompliant
+        var lambda2 = (string lambdaParam = "file://blah.txt") => lambdaParam; // Compliant
```

Comment:
Can we add some tests:
- having multiple non-compliances (with location assertion)
- combining compliant and non-compliant cases
- using a different type than `string`, but still compatible with a string literal (e.g. `IComparable uri = "a string"`

I am also wondering how the rule behaves in presence of collections, and with collection expressions:
- `IList<string> uris1 = new[] { "C:/test.txt" };`
- `IList<string> uris2 = ["C:/test.txt"];`
- `string[][] urisMatrix1 = [["C:/test.txt"]];`
- `IDictionary<string, string> urisDict = new Dictionary<string, string> { ["a"] = "C:/test.txt" };`
- ...

Reply:
Good catch! I added an issue #8172 for the collections.

Reply:
Regarding IComparable, it cannot be tested with default parameters as the default values of a reference type other than a string can only be initialized with null (therefore not raising the issue). But it's covered on `UriShouldNotBeHardcoreded_CSharp_General` 👍